src/batchpage.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import Dropdown from './dropdown.js'
+import Navbar from './navbar.js'
+import CollectionsBox from './collections-box.js'
+import Batch from './batch.js'
+import parseQuery from './parseQuery.js'
+
+class BatchPage extends React.Component {
+
+  constructor(props) {
+    super(props)
+    var args = parseQuery(window.location.href)
+    this.state = {
+      terms: args.terms ? ("" + args.terms).replace(/\+/g, ' ') : "",
+      args: args,
+      collections: [],
+      selectedCollections: [],
+    }
+  }
+
+  componentDidMount() {
+    this.getCollections()
+  }
+
+  getCollections() {
+    $.get('/collections', function (resp) {
+      var collections = resp
+      var selectedCollections = null
+      var args = this.state.args
+      if (args.collections) {
+        var sel = '' + args.collections
+        selectedCollections = sel ? sel.split('+') : []
+      }
+      else {
+        selectedCollections = resp.map((c) => c.name)
```

Comment:
I don't know how relevant it is, but perhaps it would be wise to check `resp` is the data type you expect (array).

Reply:
Eh, that's the contract of our API, if it returns 200, then it must be an array.

Reply:
I'd still put a check in place and handle the offcase when your API returns 200 but not an array. Mistakes happen.

Reply:
What should it do if it detects an error?

I'm thinking, if anything goes wrong, the UI will fail somehow, and we'll get a bug report from our kind users, and we'll fix it ASAP :)